server/src/test/java/org/elasticsearch/search/profile/AbstractProfileBreakdownTests.java

```diff
@@ -0,0 +1,144 @@
+
+        Timer firstTimerTwo = testBreakdown.getNewTimer(TestTimingTypes.TWO);
+        int firstTimerTwoCount = randomIntBetween(10, 20);
+        runTimerNTimes(firstTimerTwo, firstTimerTwoCount);
+
+        Timer secondTimerTwo = testBreakdown.getNewTimer(TestTimingTypes.TWO);
+        int secondTimerTwoCount = randomIntBetween(10, 20);
+        runTimerNTimes(secondTimerTwo, secondTimerTwoCount);
+
+        // check behaviour if one timer type hasn't been created and used
+        Map<String, Long> breakdownMap = testBreakdown.toBreakdownMap();
+        assertThat(breakdownMap.size(), equalTo(6));
+        assertThat(breakdownMap.get(TestTimingTypes.ONE.name()), equalTo(firstTimerOne.getApproximateTiming()));
+        assertThat(breakdownMap.get(TestTimingTypes.ONE.name() + "_count"), equalTo(firstTimerOne.getCount()));
+        assertThat(
+            breakdownMap.get(TestTimingTypes.TWO.name()),
+            equalTo(firstTimerTwo.getApproximateTiming() + secondTimerTwo.getApproximateTiming())
+        );
+        assertThat(breakdownMap.get(TestTimingTypes.TWO.name() + "_count"), equalTo(firstTimerTwo.getCount() + secondTimerTwo.getCount()));
+        assertThat(breakdownMap.get(TestTimingTypes.THREE.name()), equalTo(0L));
+        assertThat(breakdownMap.get(TestTimingTypes.THREE.name() + "_count"), equalTo(0L));
+
+        Timer firstTimerThree = testBreakdown.getNewTimer(TestTimingTypes.THREE);
+        int firstTimerThreeCount = randomIntBetween(10, 20);
+        runTimerNTimes(firstTimerThree, firstTimerThreeCount);
+
+        breakdownMap = testBreakdown.toBreakdownMap();
+        assertThat(breakdownMap.size(), equalTo(6));
+        assertThat(breakdownMap.get(TestTimingTypes.ONE.name()), equalTo(firstTimerOne.getApproximateTiming()));
+        assertThat(breakdownMap.get(TestTimingTypes.ONE.name() + "_count"), equalTo(firstTimerOne.getCount()));
+        assertThat(
+            breakdownMap.get(TestTimingTypes.TWO.name()),
+            equalTo(firstTimerTwo.getApproximateTiming() + secondTimerTwo.getApproximateTiming())
+        );
+        assertThat(breakdownMap.get(TestTimingTypes.TWO.name() + "_count"), equalTo(firstTimerTwo.getCount() + secondTimerTwo.getCount()));
+        assertThat(breakdownMap.get(TestTimingTypes.THREE.name()), equalTo(firstTimerThree.getApproximateTiming()));
+        assertThat(breakdownMap.get(TestTimingTypes.THREE.name() + "_count"), equalTo(firstTimerThree.getCount()));
+
+        assertThat(
+            testBreakdown.toNodeTime(),
+            equalTo(
+                firstTimerOne.getApproximateTiming() + firstTimerTwo.getApproximateTiming() + secondTimerTwo.getApproximateTiming()
+                    + firstTimerThree.getApproximateTiming()
+            )
+        );
+    }
+
+    public void testMultiThreaded() throws InterruptedException {
+        TestProfileBreakdown testBreakdown = new TestProfileBreakdown();
+        Thread[] threads = new Thread[2000];
```

Comment:
this seems a bit extreme: do we need 2k threads for this test?

Reply:
I wanted to have a high likelihood of failure for the case where we don't have synchronized access of methods / collections. I experimented with the thread count, this gave me at least a failure probability of around 30% locally. I will revise if we can lower this, I guess 10% failure likelihood would still be okay for long term testing.

Reply:
I'm reducing this to 200. This gives me approx. 20 failures out of 100 repetitions with no synchronization. Should be enough to detect any regression I think.